### DISCUSSION TITLE
Eagerly load / cache multiple entities

### DISCUSSION THREAD

**Discussion by Christoph-AK** [SeaQL/sea-orm#912] [Q&A]
Hey there, 
at the core of an app I recently finished developing is a list of recipes.
These recipes have multiple relations: CreatedBy (User), CreatedAt (User), List of ressources involved in the recipe with relative amounts.
The ressources themselves also have multiple relations: Again CreatedAt (User) and CreatedBy (User), and each a list of Tags.
For the frontend to work correctly, I need to be aware (read: include) of all the tags of each recipe, so my returned json is kinda huge. 

So far I solved this with good ol' sql, lazily querying the recipes (including two joins to the user table), and then for each recipe each of its ressources (including two joins to the user table and a trick with group_concat to get all tags for each recipe), leading to 1+n*m selects for my data. Including serialisation this sometimes leads to a ~3 second delay in calling the service, which isn't really acceptable.

I had the hope that with sea-orm I could leverage eager loading to do all selects in one fast select, but so far in trying it out it seems that sea-orm is hard limited to one relation per select. So I couldn't even combine the recipes + user1 + user 2 into one eager select.

Is this correct? Is there a workaround, apart from writing my own 'do one massive query and iterate over it, keeping track of changing IDs to identify relations', which sounds like a horrible idea, maintanance-wise?

**Comment by billy1624** [comment]
Hey @Christoph-AK, sorry for the delay!

I think this is more of a question on program design. Why not breaking the massive selects into three API endpoint? Given a known recipe ID we can get 3 different data from 3 APIs:

1. info of recipe: created by and created at
2. resources of recipe (ingredients)
3. tags of resources (ingredients)

This way API requests can be run in parallel and result would be relatively small in size
